webapp/decorators.py

```diff
@@ -24,16 +24,67 @@ def is_user_logged_in(*args, **kwargs):
     return is_user_logged_in
 
 
-def store_maintenance(func):
-    """
-    Decorator that checks if the maintence mode is enabled
-    """
+def advantage_checks(check_list=None):
+    if check_list is None:
+        check_list = []
 
-    @functools.wraps(func)
-    def is_store_in_maintenance(*args, **kwargs):
-        if strtobool(os.getenv("STORE_MAINTENANCE")):
-            return flask.render_template("advantage/maintenance.html")
+    def advantage_checks_decorator(func):
+        @functools.wraps(func)
+        def check_advantage(*args, **kwargs):
+            if "is_maintenance":
+                if strtobool(os.getenv("STORE_MAINTENANCE")):
+                    return flask.render_template("advantage/maintenance.html")
 
-        return func(*args, **kwargs)
+            is_test_backend = flask.request.args.get("test_backend", False)
+
+            stripe_publishable_key = os.getenv(
+                "STRIPE_LIVE_PUBLISHABLE_KEY",
+                "pk_live_68aXqowUeX574aGsVck8eiIE",
+            )
+
+            api_url = flask.current_app.config["CONTRACTS_LIVE_API_URL"]
+
+            if is_test_backend:
+                stripe_publishable_key = os.getenv(
+                    "STRIPE_TEST_PUBLISHABLE_KEY",
+                    "pk_test_yndN9H0GcJffPe0W58Nm64cM00riYG4N46",
+                )
+                api_url = flask.current_app.config["CONTRACTS_TEST_API_URL"]
```

Comment:
I suppose this test backend is using test a fake key, so people can't just get through stripe and get stuff for free